doc/array_buffer.md

```diff
@@ -59,8 +59,8 @@ static Napi::ArrayBuffer Napi::ArrayBuffer::New(napi_env env,
 - `[in] externalData`: The pointer to the external data to wrap.
 - `[in] byteLength`: The length of the `externalData`, in bytes.
 - `[in] finalizeCallback`: A function to be called when the `Napi::ArrayBuffer` is
-  destroyed. It must implement `operator()`, accept a `void*` (which is the
-  `externalData` pointer), and return `void`.
+  destroyed. It must implement `operator()` with parameters `(Napi::Env, void*)`,
+  and return `void`.
```

Comment:
Looks like we missed a couple of parameters I think the signature is

 static void FinalizeCallback(napi_env env, void* data, void* hint);


@computerquip any chance you can update to include the last parameter as well

Reply:
@mhdawson:
From N-API `napi_finalizer` function pointer type always has 3 parameter, but in `node-addon-api` it's different.

- for `ArrayBuffer::New(..., finalizeCallback, finalizeHint)` we have Finalizer `(Env, void* data, Hint* hint)`, like `napi_finalizer`
- for `ArrayBuffer::New(..., finalizeCallback)` we have Finalizer `(Env, void* data)`

```cpp
template <typename T, typename Finalizer, typename Hint = void>
struct FinalizeData {
  static inline
  void Wrapper(napi_env env, void* data, void* finalizeHint) {
    FinalizeData* finalizeData = static_cast<FinalizeData*>(finalizeHint);
    finalizeData->callback(Env(env), static_cast<T*>(data));
    delete finalizeData;
  }

  static inline
  void WrapperWithHint(napi_env env, void* data, void* finalizeHint) {
    FinalizeData* finalizeData = static_cast<FinalizeData*>(finalizeHint);
    finalizeData->callback(Env(env), static_cast<T*>(data), finalizeData->hint);
    delete finalizeData;
  }

  Finalizer callback;
  Hint* hint;
};
```